ci-operator/step-registry/cucushift/installer/check/vsphere/cpms/cucushift-installer-check-vsphere-cpms-commands.sh

```diff
@@ -0,0 +1,90 @@
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+
+function checkMultizone() {
+    check_result=0
+    cpms_network=$(oc get controlplanemachineset -n openshift-machine-api -ojson |  jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.network.devices')
+    cpms_workspace=$(oc get controlplanemachineset -n openshift-machine-api -ojson |  jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace')
+    cpms_template=$(oc get controlplanemachineset -n openshift-machine-api -ojson |  jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.template')
+#check network,workspace and template field in cpms.
+    if [[ ${cpms_network} == "null" ]] && [[ ${cpms_workspace} == "{}" ]] && [[ ${cpms_template} == "" ]]; then
+        echo "INFO: For multi failure domains The network,workspace and template under cpms are empty, that's expected"
+    else
+	echo "ERROR: For multi failure domains The network,workspace and template under cpms should be empty. network is ${cpms_network},workspace is ${cpms_workspace},template is ${cpms_template}" 
+        check_result=1
+    fi
+    
+#check template in infrastructure should be populated.
+    infra_template=$(oc get infrastructure -n openshift-machine-api -ojson | jq -r '.items[].spec.platformSpec.vsphere.failureDomains[].topology.template')
+    if [[ -n ${infra_template} ]]; then
+	echo "INFO: The template is populated. template under infrastructure is ${infra_template}"
+    else
+	echo "ERROR: The template is empty, please check"
+        check_result=1
+    fi
+#check the name of failuredomains in cpms should be same with infrastructure.
+    fd_name_cpms=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.failureDomains.vsphere[].name')
+    fd_name_infra=$(oc get infrastructure -n openshift-machine-api -ojson | jq -r '.items[].spec.platformSpec.vsphere.failureDomains[].name' | head -n3)
+    if [[ ${fd_name_cpms} == "${fd_name_infra}" ]]; then
+	echo "INFO: The failure domain name are same between infrastructure and cmps"
+    else
+	echo "ERROR: The failure domain name are different between infrastructure and cmps. please check"
+        check_result=1
+    fi
+    return ${check_result}
+}
+
+
+function checksinglezone() {
+        check_result=0
+	node_name=$(oc get machines.machine.openshift.io -n openshift-machine-api --selector machine.openshift.io/cluster-api-machine-type=master --no-headers | head -n1 |awk '{print $1}')
+	machine_workspace=$(oc get machine ${node_name} -n openshift-machine-api -o json |jq -r '.spec.providerSpec.value.workspace')
+	machine_template=$(oc get machine ${node_name} -n openshift-machine-api -o json |jq -r '.spec.providerSpec.value.template')
```

Comment:
the node_name is a number of nodes, can we run the one-line command against them?

Reply:
here i use head -n1, only get the first master node.

Reply:
ok